sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -645,11 +646,16 @@ object CombineConcats extends Rule[LogicalPlan] {
       stack.pop() match {
         case Concat(children) =>
           stack.pushAll(children.reverse)
+        case Cast(Concat(children), StringType, _) =>
```

Comment:
Add a comment here for the future reviewers?

Reply:
Do we need to update line 662? Could you also add a test case?

Reply:
What do we need to update in line 662?

Reply:
Ah, I see. But, IIUC there is no chance where all the children of `Concat` have `Cast`s. If all inputs of `Concat` are binary, the parent `Concat` just concats binary inputs into a binary output (In this case, children need no `Cast`).